rust/src/passwd.rs

```diff
@@ -165,6 +170,112 @@ pub fn migrate_group_except_root(rootfs_dfd: i32, preserved_groups: &Vec<String>
     Ok(())
 }
 
+/// Merge and deduplicate entries from /usr/etc and /usr/lib.
+pub fn concat_fs_content(
+    rootfs_dfd: i32,
+    mut ffi_repo: Pin<&mut crate::ffi::OstreeRepo>,
+    previous_checksum: &str,
+) -> Result<()> {
+    anyhow::ensure!(!previous_checksum.is_empty(), "missing previous reference");
+
+    let rootfs = ffiutil::ffi_view_openat_dir(rootfs_dfd);
+    let repo = ffi_repo.gobj_wrap();
+    let (prev_root, _name) = repo.read_commit(previous_checksum, gio::NONE_CANCELLABLE)?;
+
+    concat_passwd_files(&rootfs, &prev_root).context("failed to merge entries into /etc/passwd")?;
+    concat_group_files(&rootfs, &prev_root).context("failed to merge entries into /etc/group")?;
```

Comment:
Any reason we don't dedupe these functions more as was done on the C side?

Reply:
I had both versions, and I wasn't sure the increased noise was worth it. Plus, the C side used to have this comment: https://github.com/coreos/rpm-ostree/blob/v2021.1/src/libpriv/rpmostree-passwd-util.cxx#L761-L762

I added back a fixup commit with most logic deduplicated. Let me know which version you prefer to land.